src/nodes.coffee

```diff
@@ -5348,34 +5353,61 @@ unfoldSoak = (o, parent, name) ->
   ifn
 
 # Constructs a string or regex by escaping certain characters.
-makeDelimitedLiteral = (body, options = {}) ->
-  body = '(?:)' if body is '' and options.delimiter is '/'
+makeDelimitedLiteral = (body, {delimiter: delimiterOption, escapeNewlines, double, includeDelimiters = yes, escapeDelimiter = yes} = {}) ->
+  body = '(?:)' if body is '' and delimiterOption is '/'
+  escapeTemplateLiteralCurlies = delimiterOption is '`'
   regex = ///
       (\\\\)                               # Escaped backslash.
     | (\\0(?=[1-7]))                       # Null character mistaken as octal escape.
-    | \\?(#{options.delimiter})            # (Possibly escaped) delimiter.
-    | \\?(?: (\n)|(\r)|(\u2028)|(\u2029) ) # (Possibly escaped) newlines.
+    #{
+      if escapeDelimiter
+        "|\\\\?(#{delimiterOption})"       # (Possibly escaped) delimiter.
+      else
+        ''
+    }
+    #{
+      if escapeTemplateLiteralCurlies
+        "|\\\\?(\\$\\{)"                   # `${` inside template literals must be escaped.
```

Comment:
Is there a nicer pattern for being able to still use heregex/regex syntax for a conditionally included piece of a heregex (rather than having to do eg double escaping of a string that's going to be included as part of the regex like this)?

Reply:
`/\n/.source` and `/// \n ///.source` I think